auth/authenticator.hh

```diff
@@ -129,6 +134,18 @@ public:
     ///
     virtual future<custom_options> query_custom_options(std::string_view role_name) const = 0;
 
+    ///
+    /// Check if the authenticator uses salted hashes.
+    ///
+    virtual bool uses_salted_hashes() const = 0;
+
+    ///
+    /// Query the salted hash corresponding to a given role.
+    ///
+    /// If the authenticator doesn't use salted hashes, throws an `unsupported_authentication_operation` exception.
+    ///
+    virtual future<std::optional<sstring>> query_salted_hash(std::string_view role_name) const = 0;
```

Comment:
maybe default impl (indicating no support) can be added here?

i.e. `virtual bool uses_salted_hashes() const {return false}`

Reply:
Good idea. Do you think it would be also fine to add the default implementation of `query_password_hash()` that throws an exception?

Reply:
i'm ok with that